cpp/mrc/src/internal/pipeline/pipeline_instance.hpp

```diff
@@ -51,7 +51,7 @@ class PipelineInstance final : public Service, public PipelineResources
     // we need to stage those object that are created into some struct/container so we can mass start them after all
     // object have been created
     void create_segment(const SegmentAddress& address, std::uint32_t partition_id);
-    void stop_segment(const SegmentAddress& address);
+    void stop_segment(const SegmentAddress& address, bool kill = false);
```

Comment:
Consider adding a new `kill_segment` method instead of adding functionality to stop.